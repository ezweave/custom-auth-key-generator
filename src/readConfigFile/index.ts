import { readConfigFile } from './readConfigFile';

export { readConfigFile };
