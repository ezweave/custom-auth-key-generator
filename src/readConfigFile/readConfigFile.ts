import { readFileSync } from 'fs';
import yaml from 'js-yaml';
import { flow, partialRight } from 'lodash';

import { Config, parseConfig } from '../config';

export interface ReadConfigFile {
    (pathToConfigFile: string): Config 
}

export const readConfigFile: ReadConfigFile = flow(
    partialRight(
        readFileSync,
        'utf8'
    ),
    yaml.load,
    parseConfig
);
