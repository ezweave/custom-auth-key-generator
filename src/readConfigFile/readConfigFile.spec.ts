import { readConfigFile } from './readConfigFile';

describe('readConfigFile', () => {
    it('reads a valid config file and parses what is there', () => {
        expect(
            readConfigFile('./test/example.yml')
        ).toMatchSnapshot();
    });
    it('throws errors if values are missing', () => {
        expect(
            () => readConfigFile('./test/bad_example.yml')
        ).toThrowErrorMatchingSnapshot();
    });
    it('throws errors if the file isnt yaml', () => {
        expect(
            () => readConfigFile('./test/.example.env')
        ).toThrowErrorMatchingSnapshot();
    });
});
