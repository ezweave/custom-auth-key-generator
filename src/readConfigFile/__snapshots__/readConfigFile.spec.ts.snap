// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readConfigFile reads a valid config file and parses what is there 1`] = `
Object {
  "headerPrefix": "Prefix value",
  "privateKey": "MIIBVAIB=",
  "publicKey": "MFwwDQYJK==",
}
`;

exports[`readConfigFile throws errors if the file isnt yaml 1`] = `
"Missing value for required field 'headerPrefix'/'HEADER_PREFIX'
Missing value for required field 'privateKey'/'PRIVATE_KEY'
Missing value for required field 'publicKey'/'PUBLIC_KEY'"
`;

exports[`readConfigFile throws errors if values are missing 1`] = `
"Missing value for required field 'headerPrefix'/'HEADER_PREFIX'
Missing value for required field 'privateKey'/'PRIVATE_KEY'"
`;
